File: LC/lc283/lc283.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        p = 0
        for i in range(len(nums)):
            if not nums[i]:
                continue
            nums[p] = nums[i]
            if i > p:
                nums[i] = 0
            p += 1
            

File: LC/lc283/lc283.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
  let p = 0;
  for (i = 0; i < nums.length; i++){
      if (nums[i]){
          nums[p] = nums[i];
          p+=1;
      }
  }
  for (j = p; j < nums.length; j++){
      nums[j] = 0;
  }
};